nixos/modules/services/networking/nebula.nix

```diff
@@ -234,8 +258,12 @@ in
 
     # Open the chosen ports for UDP.
     networking.firewall.allowedUDPPorts =
-      unique (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks);
-
+      unique (flatten (mapAttrsToList (netName: netCfg: [netCfg.listen.port] ++ optional (netCfg.serveDns) netCfg.dns.port) enabledNetworks));
+assertions =
+      [ { assertion = !(!netcfg.isLighthouse && netcfg.serverDns)
```

Comment:
I think we should add a dns.openFirewall boolean flag that enables this behavior.